### ISSUE TITLE
building netgen with occ 7.5.0 on windows - cmake failure

### ISSUE THREAD

**Issue by kcris** [NGSolve/netgen#104]
Hi
I am building OCC 7.5.0 on windows 10, no issues whatsoever.
Then I use latest netgen master (sha=d7be2c8), and I try to build like this

```
cmake^
	-DZLIB_LIBRARY=%MY_ZLIB_DIR%/lib/zlib.lib^
	-DZLIB_INCLUDE_DIR=%MY_ZLIB_DIR%/include^
	-DOCC_INCLUDE_DIR=%MY_OCE_INC_DIR%^
	-DOCC_LIBRARY_DIR=%MY_OCE_LIB_DIR_LINK%^
	-DOCC_LIBRARY=%MY_OCE_LIB_DIR_LINK%/TKernel.lib^
	-DOCC_VERSION_STRING=7.5.0FF^
	-DUSE_SUPERBUILD=OFF^
	-DUSE_PYTHON=OFF^
	-DUSE_GUI=OFF^
	-DUSE_OCC=ON^
	-DCMAKE_CONFIGURATION_TYPES:STRING=Debug;Release;RelWithDebInfo^
	-DBUILD_SHARED_LIBS=OFF^
	-DCMAKE_STATIC_LINKER_FLAGS="/LIBPATH:%FF_THIRDPARTIES_INSTALL_DIR%\lib"^
	-G "%FF_ARG_CMAKE_GENERATOR%" -A "%FF_ARG_CMAKE_ARCH%"^
	%MYSRCDIR%
```

and a bunch of occ cmake files are referenced but not found

![image](https://user-images.githubusercontent.com/812733/145284719-ab93e4b7-d6cc-4fda-903e-5b24cd0e6022.png)

Those files do exist somewhere inside the OCC build dir (`oce/CMakeFiles/Export/cmake`)
but I have no idea how exactly netgen expects to find those.

I have naively tried to copy those `OpenCascade... .cmake` files to current folder but that results in more and more problems.

Am I missing some OCC variable I have to pass to netgen?

Thanks

ps: I did not had the issue while using Netgen tag `v6.2.2104` (July 6th)
so I guess the problem is in a recent commit

**Comment by mhochsteger** [comment]
Hi,

Seems that some OCC modules are missing in your build. How did you configure Opencascade?
The reason for you issue is that OCC is now found by cmake using
`find_package(OpenCascade ...)` (see https://github.com/NGSolve/netgen/blob/master/CMakeLists.txt#L331 ).

Note that the Netgen superbuild supports building a static version of OCC using `-DBUILD_OCC=ON`. After building, it passes `OpenCascade_ROOT` to the actual Netgen-Cmake call. Since you are not using the superbuild, you also might want to compare your OCC configuration with the one we are using: https://github.com/NGSolve/netgen/blob/master/cmake/SuperBuild.cmake#L102

Best,
Matthias

**Comment by kcris** [comment]
H @mhochsteger , 

First of all thanks for your response.

Yes, Netgen wants the opencascade .cmake files. And they are present, in various folders.
Yes, I have a static version of occ (7.6.0).

The toplevel OCC folder only contains these .cmake files
```bash
OpenCASCADECompileDefinitionsAndFlags-debug.cmake
OpenCASCADECompileDefinitionsAndFlags-release.cmake
OpenCASCADECompileDefinitionsAndFlags-relwithdebinfo.cmake
OpenCASCADEConfig.cmake
OpenCASCADEConfigVersion.cmake
OpenCASCADETargets.cmake
```
![image](https://user-images.githubusercontent.com/812733/145483249-5f539f99-8d8c-4a08-a887-c13aa392a635.png)


The OTHER opencascade .cmake files are present but *not* in the same folder as the files above.
They are in the `CMakeFiles/Export/cmake` subdir, this is how OCCT has created them.

![image](https://user-images.githubusercontent.com/812733/145483348-9a2df883-3065-4ed5-a9e9-1ca02d087f9a.png)


And Netgen will not find them there:

I try to build Netgen like this
```bash
cmake^
	-DZLIB_LIBRARY=%MY_ZLIB_DIR%/lib/zlib.lib^
	-DZLIB_INCLUDE_DIR=%MY_ZLIB_DIR%/include^
	-DOpenCascade_INCLUDE_DIR=%MY_OCCT_INC_DIR%^
	-DOpenCascade_LIBRARY_DIR=%MY_OCCT_LIB_DIR_LINK%^
	-DOpenCascade_LIBRARY=%MY_OCCT_LIB_DIR_LINK%/TKernel.lib^
	-DOPENCASCADE_VERSION_STRING=7.6.0FF^
	-DUSE_SUPERBUILD=OFF^
	-DUSE_PYTHON=OFF^
	-DUSE_GUI=OFF^
	-DUSE_OCC=ON^
	-DCMAKE_CONFIGURATION_TYPES:STRING=Debug;Release;RelWithDebInfo^
	-DBUILD_SHARED_LIBS=OFF^
	-DCMAKE_STATIC_LINKER_FLAGS="/LIBPATH:%FF_THIRDPARTIES_INSTALL_DIR%\lib"^
	-G "%FF_ARG_CMAKE_GENERATOR%"^
        -A "%FF_ARG_CMAKE_ARCH%"^
	%MYSRCDIR%
```
* netgen sees those 6 cmake files that *are* available in the toplevel OCC folder (e.g. `OpenCASCADEConfig.cmake`)
* but gets stuck with all the other .cmake files that are *not* here, but only present in that `CMakeFiles/Export/cmake` subdirectory (e.g. `OpenCASCADEDrawTargets.cmake`, `OpenCASCADEModelingDataTargets.cmake`...)

I tried to move those separate .cmake files, manually, 
to the expected location (next to the `OpenCASCADEConfig.cmake`), and I get other issues.

So I guess that 
* I am either the only one compiling on windows (both occ and netgen) 
* or I am missing some other fundamental param that I have to pass to cmake

Any clues? 
Thank you

**Comment by kcris** [comment]
tested with OCCT 7.5.0 and OCCT 7.6.0, in our case netgen fails to find Opencascade so cmake fails

these two commits are causing the problem
* https://github.com/NGSolve/netgen/commit/40daa032
* https://github.com/NGSolve/netgen/commit/467d0e56

a git revert fixes the build problem

thanks to @mostafanou7

**Comment by kcris** [comment]
found the root of the problem

I use opencascade and netgen built in multiple configurations: Debug. Release. ReleaseWithDebugInfo

The cmake `--install` target of opencascade has some issues: it only handles correctly the Release configuration 
(for Debug and ReleaseWithDebugInfo the library files are not copied to the installation folder).

Because of that, I was not using cmake --install for opencascade
Instead, I was copying opencascade libs+headers manually. 
Except that by doing so, some cmake files used with FindPackage were not copied.
And that resulted in Netgen being unable to find those .cmake files which means Netgen was not building.

The solution: switched back to a mixture of 
* cmake --install 
* some additional copying (for the missing libs) to the opencascade install dir
Passing that install dir to Netgen works now just fine with latest netgen (tested with occt 7.5.0 / 7.6.0)

Long story short: it was an opencascade issue with incomplete handling of the --install cmake target.

Thanks, closing
